api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -252,6 +254,53 @@ default boolean createFolder( @Nullable String path ) throws OperationFailedExce
+   * allowed.
+   *
+   * @param path    The path of the generic file to create.
+   * @param content The content to write to the file as an InputStream.
+   * @return {@code true}, if the file did not exist and was created; {@code false}, if the file already existed.
+   * @throws AccessControlException    If the current user cannot perform this operation.
+   * @throws InvalidPathException      If the file path is not valid.
+   * @throws InvalidOperationException If the path, or one of its prefixes, does not exist and cannot be created
+   *                                   using this service (e.g. connections, buckets);
+   *                                   if the path or its longest existing prefix does not reference a folder;
+   *                                   if the path does not exist and the current user is not allowed to create
+   *                                   files on the folder denoted by its longest existing prefix.
+   * @throws OperationFailedException  If the operation fails for some other (checked) reason.
+   */
+  boolean createFile( @NonNull GenericFilePath path, boolean overwrite, @NonNull InputStream content )
+    throws OperationFailedException;
+
+  /**
+   * Creates a file given its path and content, with overwrite option.
+   * <p>
+   * This method ensures that each ancestor folder of the specified file exists, creating it if necessary, and
+   * allowed.
+   *
+   * @param createFileOptions The options for creating the file, including path, content and overwrite flag.
+   * @return {@code true}, if the file was created or overwritten; {@code false}, if the file already existed and
+   * overwrite is false.
+   * @throws AccessControlException    If the current user cannot perform this operation.
+   * @throws InvalidPathException      If the file path is not valid.
+   * @throws InvalidOperationException If the path, or one of its prefixes, does not exist and cannot be created
+   *                                   using this service (e.g. connections, buckets);
+   *                                   if the path or its longest existing prefix does not reference a folder;
+   *                                   if the path does not exist and the current user is not allowed to create
+   *                                   files on the folder denoted by its longest existing prefix.
+   * @throws OperationFailedException  If the operation fails for some other (checked) reason.
+   */
+  default boolean createFile( @NonNull String path,
+                              @NonNull CreateFileOptions createFileOptions,
+                              @NonNull InputStream content )
+    throws OperationFailedException {
+    // Default implementation for backward compatibility - delegates to existing method
+    return createFile( GenericFilePath.parseRequired( path ), createFileOptions.isOverwrite(), content );
```

Comment:
Consider adding a couple of tests to this. Check out the existing, similar `CreateFolderTests`.

Reply:
addressed